budgetportal/webflow/views.py

```diff
@@ -60,3 +62,30 @@ class ProvInfraProjectView(generics.ListAPIView):
         "program_implementing_agent",
         "other_parties",
     ]
+
+    def list(self, request, *args, **kwargs):
+        base_url = (
+            get_current_site(request).domain + "/infrastructure-projects/provincial/"
+        )
+        project_urls = []
+
+        queryset = self.filter_queryset(self.get_queryset())
+        for project in queryset:
+            project_urls.append(project.get_url_path())
```

Comment:
I don't think you should be looping over values here - see https://stackoverflow.com/questions/18396547/django-rest-framework-adding-additional-field-to-modelserializer

Looping over values would destroy much of the benefit of paging.

Reply:
overridden method deleted

https://github.com/vulekamali/datamanager/pull/190/commits/2e5807c2e032a6dad9ea54dd0c5ecd5d23d11ba8